### PR TITLE
test: add tests for dev server and proxy

### PR DESCRIPTION
**Pull request by manniL**
### 🔗 Linked issue

#641 - not resolving though

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Initial plan was to check why the proxy is failing. In the local setup, the proxy did not fail.
I've added some test for the proxy and for the dev preset in general to ensure

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: test/tests.ts [hunk 1]
```diff
@@ -63,11 +71,21 @@ export async function setupTest (preset) {
...
   console.log('>', ctx.server!.url)
 }
 
+export async function startDevServer (ctx: Context) {
+  const devServer = createDevServer(ctx.nitro)
+  ctx.server = await devServer.listen({})
+  await prepare(ctx.nitro)
+  await build(ctx.nitro)
+  // TODO: How to await that dev server is ready?!
+  await new Promise(resolve => setTimeout(resolve, 1000))
```
Comment (manniL) [inline test/tests.ts:83]: Did not find a better way 🙈

File: test/tests.ts [hunk 2]
```diff
@@ -63,11 +69,21 @@ export async function setupTest (preset) {
...
```
Comment (danielroe) [inline test/tests.ts:83]: wdyt @pi0?

```suggestion
  const ready = new Promise<void>((resolve) => { ctx.nitro.hooks.hook('dev:reload', () => resolve()) })
  await build(ctx.nitro)
  await ready
```

Comment (pi0) [reply test/tests.ts:83]: `dev:reload` hook seems a nice idea! Addtionally we need to move this test to dev preset.

Comment (danielroe) [reply test/tests.ts]: @manniL would you move this to the dev preset? I think @pi0 just means extracting this function and inlining it in the dev preset test file.

Comment (manniL) [reply test/tests.ts]: Ohh, yes sure 👍🏻

**Review by danielroe — APPROVED** [review]

**Review by pi0 — APPROVED** [review]
Thanks!

**Pull request merged**